apps/src/gamelab/GameLabSprite.js

```diff
@@ -322,9 +322,9 @@ var _collideWith = function (p5Inst, type, target, callback) {
   if (state.__i < state.__others.length) {
     state.__result = this._collideWithOne(type, state.__others[state.__i], callback) || state.__result;
     // Not done, unless we're on the last item in __others:
-    state.doneExec = state.__i >= (state.__others.length - 1);
+    state.doneExec_ = state.__i >= (state.__others.length - 1);
   } else {
-    state.doneExec = true;
+    state.doneExec_ = true;
   }
   return state.__result;
```

Comment:
So this code is actually covered by tests, but there aren't proper assertions to insure that it actually works. I'd like to actually move this code out of gamelab and into the jsinterpreter directory so we can keep all the interpreter shenanigans in one place.

I'm wondering what the generic feature is here. Make an interpreter function loop based on a native predicate?